Yosemite/Yosemite/Stores/ProductStore.swift

```diff
@@ -1162,25 +1162,25 @@ extension ProductStore {
     /// Updates, inserts, or prunes the provided `storageProduct`'s custom fields using the provided `readOnlyProduct`'s custom fields
     ///
     private func handleProductCustomFields(_ readOnlyProduct: Networking.Product, _ storageProduct: Storage.Product, _ storage: StorageType) {
-        // Upsert the `customFields` from the `readOnlyProduct`
-        readOnlyProduct.customFields.forEach { readOnlyCustomField in
-            if let existingStorageMetaData = storage.loadProductMetaData(siteID: readOnlyProduct.siteID,
-                                                                         productID: storageProduct.productID,
-                                                                         metadataID: readOnlyCustomField.metadataID) {
-                existingStorageMetaData.update(with: readOnlyCustomField)
-            } else {
-                let newStorageMetaData = storage.insertNewObject(ofType: Storage.MetaData.self)
-                newStorageMetaData.update(with: readOnlyCustomField)
-                storageProduct.addToCustomFields(newStorageMetaData)
-            }
-        }
+        // Create a set of metadata IDs from the read-only product for quick lookup
+        let readOnlyMetadataIDs = Set(readOnlyProduct.customFields.map { $0.metadataID })
```

Comment:
I think this isn't used anywhere.

Reply:
d51ee596fb1c3c0533cacb338e2eb755c3c3bb8e